packages/ember-glimmer/lib/environment.js

```diff
@@ -79,14 +79,20 @@ export default class Environment extends GlimmerEnvironment {
     return false;
   }
 
-  getComponentDefinition(name) {
+  getComponentDefinition(nameSegments, meta) {
+    let name = nameSegments.join('/');
     let definition = this._components[name];
 
     if (!definition) {
-      let { component: ComponentClass, layout } = lookupComponent(this.owner, name[0]);
+      let { component: ComponentClass, layout } = lookupComponent(this.owner, name);
```

Comment:
To support local lookup we need to pass a third arg to `lookupComponent` here ([see here](https://github.com/emberjs/ember.js/blob/master/packages/ember-glimmer/lib/utils/lookup-component.js#L27)). I think that arg would be `{ source: meta && meta.moduleName}`.  Then that helper function looks for a local result first, and falls back to global.  I'm not sure why we need `else if` below if we pass the third arg to `lookupComponent`.